src/viam/ur/module/ur_arm_state.cpp

```diff
@@ -285,11 +288,20 @@ URArm::state_::arm_connection_::~arm_connection_() {
     dashboard.reset();
 }
 
-URArm::state_::move_request::move_request(std::vector<trajectory_sample_point>&& samples, std::ofstream arm_joint_positions_stream)
-    : samples(std::move(samples)), arm_joint_positions_stream(std::move(arm_joint_positions_stream)) {
-    if (this->samples.empty()) {
-        throw std::invalid_argument("no trajectory samples provided to move request");
-    }
+URArm::state_::move_request::move_request(MoveCommand&& move_command, std::ofstream arm_joint_positions_stream)
+    : move_command(std::move(move_command)), arm_joint_positions_stream(std::move(arm_joint_positions_stream)) {
+    // Validate the move command based on its type
+    std::visit(
+        [](const auto& cmd) {
+            using T = std::decay_t<decltype(cmd)>;
+            if constexpr (std::is_same_v<T, std::vector<trajectory_sample_point>>) {
+                if (cmd.empty()) {
+                    throw std::invalid_argument("no trajectory samples provided to move request");
```

Comment:
This also argues against your helper function which seems to exist mostly to do this same check.